### ISSUE TITLE
Hash navigation not working [TypeError: normalized[thumbIndex] is undefined]

### ISSUE THREAD

**Issue by TiraelSedai** [dromru/react-photoswipe-gallery#276]
When I open root initially, it works fine.

However if I try to come to specific slide externally (e.g. I want to share it), it fails to do so.

**To reproduce**:
```
<Gallery id="1">
... several Item s ...
</Gallery>
```
Go to: http://localhost:3000/#&gid=1&pid=12

**Expected**: Item with id = 12 is opened

**Actual**:
```
TypeError: normalized[thumbIndex] is undefined
getThumbBoundsFn
c:/git/src/gallery-custom.tsx:93

  90 |         .sort(function (_a, _b) {
  91 |         var _c = __read(_a, 1), a = _c[0].current;
  92 |         var _d = __read(_b, 1), b = _d[0].current;
> 93 |         return helpers_1.sortNodes(a, b);
  94 | ^    })
  95 |         .forEach(prepare);
  96 | }
```

```

./node_modules/react-photoswipe-gallery/dist/gallery-custom.js/exports.CustomGallery/open<
c:/git/src/gallery-custom.tsx:103

  100 | var layoutEl = layoutRef.current;
  101 | if (layoutEl) {
  102 |     var instance = new photoswipe_1.default(layoutEl, ui, normalized, __assign(__assign({ index: index === null ? parseInt(targetId, 10) - 1 : index, getThumbBoundsFn: function (thumbIndex) {
> 103 |             var el = normalized[thumbIndex].el;
  104 | ^           return el ? helpers_1.getElBounds(el) : { x: 0, y: 0, w: 0 };
  105 |         }, history: false }, (galleryUID !== undefined
  106 |         ? { galleryUID: galleryUID, history: true }
```

**Comment by TiraelSedai** [comment]
Related #21

**Comment by irudoy** [comment]
Thanks for reporting! Seems like hash navigation is broken. I'll investigate into it soon

**Comment by nohardcode** [comment]
I can't reproduce the problem. Hash navigation works well for me with following code.

```
import React from 'react';
import { Gallery, Item } from 'react-photoswipe-gallery';
import 'photoswipe/dist/photoswipe.css'
import 'photoswipe/dist/default-skin/default-skin.css'

function App() {
  return (
    <Gallery id="1">
      <Item
        original="https://placekitten.com/1024/768?image=1"
        thumbnail="https://placekitten.com/160/120?image=1"
        width={1024}
        height={768}
        id={1}
      >
        {({ ref, open }) => ( 
            <img
              onClick={open}
              src="https://placekitten.com/160/120?image=1"
              ref={ref}
            />   
        )}
      </Item>
      <Item
        original="https://placekitten.com/1024/768?image=2"
        thumbnail="https://placekitten.com/160/120?image=2"
        width={1024}
        height={768}
        id={2}
      >
        {({ ref, open }) => (
            <img
              onClick={open}
              src="https://placekitten.com/160/120?image=2"
              ref={ref}
            />
        )}
      </Item>
      <Item
        original="https://placekitten.com/1024/768?image=3"
        thumbnail="https://placekitten.com/160/120?image=3"
        width={1024}
        height={768}
        id={3}
      >
        {({ ref, open }) => (
            <img
              onClick={open}
              src="https://placekitten.com/160/120?image=3"
              ref={ref}
            />
        )}
      </Item>
    </Gallery>
  );
}

export default App;
```

@TiraelSedai  can you please provide more context on this? Maybe there is a very specific case, which I cannot reproduce

**Comment by TiraelSedai** [comment]
@nohardcode Hi, it appears that indeed your example works and the issue is with the other component.
I use https://github.com/tsuyoshiwada/react-stack-grid this greed between Gallery and Items inside it, and if I remove it (or change to a simple div), I see that navigation is working.

My apologies, it appears that the issue is elsewhere (I just saw the crash in RPG lib), however I'm not sure how I'd resolve it.

**Comment by TiraelSedai** [comment]
@nohardcode 

Hi again. I'm not sure if I need to reopen or just create new issue just asking for advice, so I'll just try to ask here first.

So the issue persists when I use other React components as layout between Gallery component and Item s components.

At first I thought that issue is with https://github.com/tsuyoshiwada/react-stack-grid however I've tried https://github.com/dantrain/react-stonecutter and I see exact same issue there.

Do you have an advice on how do I debug this? I really want a waterfall pinterest-like layout for my Photoswipe.

**Comment by irudoy** [comment]
@TiraelSedai it would be great if you provided an isolated repro for it

**Comment by TiraelSedai** [comment]
https://github.com/TiraelSedai/rpg-hash-nav-issue-example sure thing

**Comment by irudoy** [comment]
fixed in #300
